src/rect.rs

```diff
@@ -768,4 +824,54 @@ mod tests {
         let test = Rect::new(0.0, 0.0, 1.0, 1.0);
         assert!((test.aspect_ratio() - 1.0).abs() < 1e-6);
     }
+
+    #[test]
```

Comment:
Maybe consider adding a test for (0, 0, 10, 10), (0, 10, 10, 20) (to demonstrate that rects overlap if they share a boundary.